front/src/Stores/BiggestAvailableAreaStore.ts

```diff
@@ -123,4 +97,39 @@ function createBiggestAvailableAreaStore() {
     };
 }
 
+export function findAllFreeBoxes(left: number, top: number, grid: boolean[][]): Box[] {
+    if (grid.length === 0) {
+        return [];
+    }
+    if (grid[0].length === 0) {
+        return [];
+    }
+    if (grid[top][left] === true) {
+        return [];
+    }
+
+    const squares: Box[] = [];
+
+    // we expect grid rows to be of the same length
+    let xEnd = grid[0].length;
+
+    for (let y = top; y < grid.length; y += 1) {
+        for (let x = left; x < xEnd; x += 1) {
+            // do not add squares with occupied parts
+            if (grid[y][x] === true) {
+                // no point in trying to find free square after this column
+                xEnd = x;
+                break;
+            }
+            squares.push({ xStart: left, yStart: top, xEnd: x, yEnd: y });
+        }
+    }
```

Comment:
```suggestion
    for (let y = top; y < grid.length; y += 1) {
        for (let x = left; x < xEnd; x += 1) {
            // do not add squares with occupied parts
            if (grid[y][x] === true) {
                // no point in trying to find free square after this column
                xEnd = x;
                break;
            }
        }
        squares.push({ xStart: left, yStart: top, xEnd: xEnd, yEnd: y });
    }
```

I think you can safely add a single square at the end of the inner for loop.
Indeed, the last square of the inner loop contains all the squares defined in the inner loop.